### PR TITLE
OAK-11306: fix GCP bucket cleanup post running tests

### PR DESCRIPTION
**Pull request by t-rana**

### CODE REVIEW

File: oak-blob-cloud/src/main/java/org/apache/jackrabbit/oak/blob/cloud/s3/S3Backend.java [hunk 1]
```diff
@@ -1368,8 +1370,15 @@ public KeyRenameThread(String oldKey) {
     /**
      * Enum to indicate remote storage mode
      */
-    private enum RemoteStorageMode {
+    enum RemoteStorageMode {
         S3,
-        GCP
+        GCP;
```
Comment (rishabhdaim) [inline oak-blob-cloud/src/main/java/org/apache/jackrabbit/oak/blob/cloud/s3/S3Backend.java]:
```suggestion
                enum RemoteStorageMode {
        S3("s3"),
        GCP("gcp");

        RemoteStorageMode(String name) {
        }

        @NotNull
        static RemoteStorageMode fromString(String mode) {

            switch (mode.toLowerCase()) {
                case "s3": return S3;
                case "gcp": return GCP;
                default: return S3;
            }
        }
```

Comment (t-rana) [reply oak-blob-cloud/src/main/java/org/apache/jackrabbit/oak/blob/cloud/s3/S3Backend.java]: modified the code using this functionality, so now this is not required.

File: oak-blob-cloud/src/main/java/org/apache/jackrabbit/oak/blob/cloud/s3/S3Backend.java [hunk 2]
```diff
@@ -25,6 +25,7 @@
 import java.net.URL;
 import java.time.Instant;
 import java.util.ArrayList;
+import java.util.Arrays;
```
Comment (rishabhdaim) [inline oak-blob-cloud/src/main/java/org/apache/jackrabbit/oak/blob/cloud/s3/S3Backend.java]: remove this import

File: oak-blob-cloud/src/main/java/org/apache/jackrabbit/oak/blob/cloud/s3/S3Backend.java [hunk 3]
```diff
@@ -33,6 +34,7 @@
 import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
```
Comment (rishabhdaim) [inline oak-blob-cloud/src/main/java/org/apache/jackrabbit/oak/blob/cloud/s3/S3Backend.java]: remove this one too

**Review by rishabhdaim — CHANGES_REQUESTED** [review]

File: oak-blob-cloud/src/main/java/org/apache/jackrabbit/oak/blob/cloud/s3/S3Backend.java (hunk 2 above)
Comment (t-rana) [reply oak-blob-cloud/src/main/java/org/apache/jackrabbit/oak/blob/cloud/s3/S3Backend.java]: done

File: oak-blob-cloud/src/main/java/org/apache/jackrabbit/oak/blob/cloud/s3/S3Backend.java (hunk 3 above)
Comment (t-rana) [reply oak-blob-cloud/src/main/java/org/apache/jackrabbit/oak/blob/cloud/s3/S3Backend.java]: done

**Review by rishabhdaim — APPROVED** [review]

**Review by rishabhdaim — APPROVED** [review]

**Pull request merged**